HISTORY.txt

```diff
@@ -1,11 +1,17 @@
 Changes
 =======
 
-23.12.11.dev0 (Unreleased)
+23.12.14 (2023-12-14)
```

Comment:
This is weird. Shouldn't this be a placeholder version until it's actually released?

Reply:
@ctheune I included that to show that once #187 is merged and then subsequently this PR, #183 I intend to make this a PyPi release to get the bug fix out to end users. So adding that in as part of the release notes. Once @zware has confirmed required changes were made I'd update that version number to the current date at the time. And then release on PyPi

Reply:
Ok, not sure I can follow the reasoning.

Tooling for the release process should be updating that number automatically or with user prompt. What's your workflow for cutting a release at the moment? I used to use `zest.releaser` which had this automated so it was never touched in any specific PR.

Reply:
@ctheune shows that this is not my area of expertise 😅

For the previous two releases I: 

1. Updated the files all on main with the new version number in both `history.txt` and `pyproject.toml`
2. Deleted any existing files for old versions in my local `dist/` folder
3. Created the new `.whl` and `.tar.gz` by running the following command at the CWD
``` bash
python3 -m build
```
4. Pushing to PyPi by using
``` bash
python3 -m twine upload dist/*
username: __token__
password: api_token
```

Is there a better way? 🤣

Reply:
I assumed that before running python3 -m build everything needed to be neatly tied up.

Reply:
Yeah right, so there's automation for those things that make releases "single click" (or "single enter") and take care of all those steps consistently.

If you like I can show you how I used `zest.releaser` until now. There's other options, too.

Reply:
That's amazing! Let me do some research. It seems like for a lot of these things there's more than "one way to skin a cat" and once I know you want me to go from "head to tail" or "tail to head" I can usually do the research necessary to figure it out. Let me do some research on `zest.releaser` and I'll follow up with any questions. I'll avoid merging this PR / deploying until I'm ready to use that :)

Reply:
Another idea for renovation would be to switch from setuptools to hatch(ling?). I think their automation isn't quite as complete. However, I *think* some people also do the automation using github actions and the "make a release" button on Github. I'm just more of a fan of having all actions available on the CLI and then the potential to invoke this from actions than relying on fully integrated "magic" stuff.

Reply:
We're already off of `setuptools` entirely with `poetry` :)